onnxruntime/core/external_ops/pyop.h

```diff
@@ -0,0 +1,209 @@
+    Initialize*          initialize_             = nullptr;
+    NewInstance*         new_instance_           = nullptr;
+    InvokePythonFunc*    invoke_python_func_     = nullptr;
+    ReleaseInstance*     release_instance_       = nullptr;
+    GetLastErrorMessage* get_last_error_message_ = nullptr;
+
+private:
+
+    PyOpLibProxy() {
+        std::string err;
+        LOAD_PYOP_LIB(LIB_PYOP,              handle_,                 "Failed to load pyop library");
+        LOAD_PYOP_SYM("Initialize",          initialize_,             "Failed to import function: Initialize");
+        LOAD_PYOP_SYM("NewInstance",         new_instance_,           "Failed to import function: NewInstance");
+        LOAD_PYOP_SYM("InvokePythonFunc",    invoke_python_func_,     "Failed to import function: InvokePythonFunc");
+        LOAD_PYOP_SYM("ReleaseInstance",     release_instance_,       "Failed to import function: ReleaseInstance");
+        LOAD_PYOP_SYM("GetLastErrorMessage", get_last_error_message_, "Failed to import function: GetLastErrorMessage");
+        ORT_ENFORCE  (initialize_(),         get_last_error_message_(err));
+    }
+
+    ~PyOpLibProxy() {
+        Env::Default().UnloadDynamicLibrary(handle_);
+    }
+};
+
+struct PyCustomKernel {
+
+    PyCustomKernel(Ort::CustomOpApi   ort,
+                   const OnnxAttrs&   attrs,
+                   const std::string& module,
+                   const std::string& class_name,
+                   const std::string& compute,
+                   PyOpLogFunc        logging_func):
+                   ort_(ort), attrs_(attrs), module_(module), class_name_(class_name),
+                   compute_(compute), logging_func_(logging_func) {
+        std::string err;
+        instance_ = PyOpLibProxy::GetInstance().new_instance_(module.c_str(), class_name_.c_str(), attrs_);
+        ORT_ENFORCE(nullptr != instance_, PyOpLibProxy::GetInstance().get_last_error_message_(err));
+    }
+
+    ~PyCustomKernel() {
+        if (nullptr != instance_) {
+            PyOpLibProxy::GetInstance().release_instance_(instance_);
+            instance_ = nullptr;
+        }
+    }
+
+    // Do nothing since Custom Op does not trigger shape inference
+    void GetOutputShape(OrtKernelContext*, size_t, OrtTensorTypeAndShapeInfo*) {}
+
+    void Compute(OrtKernelContext* context) {
+
+        ORT_ENFORCE (nullptr != context);
+        auto inputs_count = (size_t)reinterpret_cast<onnxruntime::OpKernelContextInternal*>(context)->InputCount();
+        std::vector<const void*>             inputs;
+        std::vector<std::unique_ptr<char[]>> outputs;
+        std::vector<int32_t>                 inputs_type, outputs_elem_size;
+        std::vector<std::vector<int64_t>>    inputs_dim,  outputs_dim;
+
+        for (size_t i = 0; i < inputs_count; ++i) {
+            auto ort_value = ort_.KernelContext_GetInput(context, i);
+            inputs.push_back(((MLValue*)ort_value)->Get<Tensor>().DataRaw());
```

Comment:
use c++ style cast. there're many places that need this correction.

Reply:
Done.